pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1743,6 +1752,12 @@ private void updateCurrentDocumentCountMetrics() {
           rowsConsumed, consumedRate, _currentOffset, _numRowsConsumed, _numRowsIndexed);
       _lastConsumedCount = _numRowsConsumed;
       _lastLogTime = now;
+      if (_filteredMessageOffsets.size() > 0) {
+        if (_trackFilteredMessageOffsets) {
+          _segmentLogger.info("Filtered events with offsets: {}", _filteredMessageOffsets);
```

Comment:
if there are a lot of offsets, then the log line may become super high. Maybe log a sublist of 5-10 elements if the length exceeds a certain threshold?

Reply:
Hmm this sublist length may vary from user to user.. some way want 50 some may want 5-10.. how about we leave this to the log-truncation logic on the user side itself? As this log statement is behind a config, user might expect a long-line and should be relatively safe. We can revisit this later if we feel this is creating a lot of logging nuisance. Wdyt?